internal/controller/kyma_controller.go

```diff
@@ -405,21 +406,60 @@ func (r *KymaReconciler) handleDeletingState(ctx context.Context, kyma *v1beta2.
 		}
 
 		r.RemoteClientCache.Del(client.ObjectKeyFromObject(kyma))
-		if err := remote.RemoveFinalizerFromRemoteKyma(ctx, r.RemoteSyncNamespace); client.IgnoreNotFound(err) != nil {
+		if err := remote.RemoveFinalizersFromRemoteKyma(ctx, r.RemoteSyncNamespace); client.IgnoreNotFound(err) != nil {
 			err = fmt.Errorf("error while trying to remove finalizer from remote: %w", err)
 			r.enqueueWarningEvent(kyma, deletionError, err)
 			return ctrl.Result{}, err
 		}
 
-		logger.Info("removed remote finalizer")
+		logger.Info("removed remote finalizers")
 	}
 
 	r.Metrics.CleanupMetrics(kyma.Name)
+	relatedManifests, err := r.getRelatedManifestCRs(ctx, kyma)
+	if err != nil {
+		err = fmt.Errorf("error while trying to get manifests: %w", err)
+		r.enqueueWarningEvent(kyma, deletionError, err)
+		return ctrl.Result{}, err
+	}
+
+	if !r.relatedManifestCRsAreDeleted(relatedManifests) {
+		if err = r.deleteManifests(ctx, relatedManifests); err != nil {
+			r.enqueueWarningEvent(kyma, deletionError, err)
+			return ctrl.Result{}, err
+		}
+		return ctrl.Result{Requeue: true}, nil
+	}
```

Comment:
```suggestion
	relatedManifests, err := r.getRelatedManifestCRs(ctx, kyma)
	if err != nil {
		err = fmt.Errorf("error while trying to get manifests: %w", err)
		r.enqueueWarningEvent(kyma, deletionError, err)
		return ctrl.Result{}, err
	}

	if !r.relatedManifestCRsAreDeleted(relatedManifests) {
		if err = r.deleteManifests(ctx, relatedManifests); err != nil {
			r.enqueueWarningEvent(kyma, deletionError, err)
			return ctrl.Result{}, err
		}
		return ctrl.Result{Requeue: true}, nil
	}
	r.Metrics.CleanupMetrics(kyma.Name)
```

I would prefer first cleaning up manifests, before cleaning up the metrics